playbooks/common/openshift-cluster/upgrades/upgrade_control_plane.yml

```diff
@@ -234,21 +234,36 @@
 
   - when: (openshift.common.version_gte_3_6 | bool) and (not openshift.common.version_gte_3_7 | bool)
     block:
+    - name: Retrieve shared-resource-viewer
+      oc_obj:
+        state: list
+        kind: role
+        name: "shared-resource-viewer"
+        namespace: "openshift"
+      register: objout
+
+    - name: Determine if shared-resource-viewer is protected
+      set_fact:
+        __shared_resource_viewer_protected: "{{ objout['results']['results']['metadata']['annotations']['openshift.io/reconcile-protect'] }}"
+      when: "'results' in objout and 'results' in objout['results'] and 'openshift.io/reconcile-protect' in objout['results']['results'][0]['metadata']['annotations']"
```

Comment:
The value of the annotation must be true.  `openshift.io/reconcile-protect==true` means do not reconcile.

Reply:
I pushed a version that checks for that